arraysForDeletion.js

```diff
@@ -0,0 +1,246 @@
+        rightEdgesInt.push(board1.x-1) // right edge of the board always starts with x-1(the count starts from 0 not from 1, so 1 should be subtracted)
+        for (let i =0;i<board1.y;i++){
+            addNumToArray = addNumToArray+board1.x
+            if (addNumToArray<board1.size){
+                rightEdgesInt.push(addNumToArray)
+            }
+        }
+    }
+    getLeftAndRightEdges()
+
+    for (let i = 0;i<board1.size;i++){
+        tabsIntWithoutEdges.push(i)
+    }
+    for (let i=0;i<leftEdgesInt.length;i++){
+        let numberDelete = tabsIntWithoutEdges.indexOf(leftEdgesInt[i])
+        tabsIntWithoutEdges.splice(numberDelete,1)
+    }
+    for (let i=0;i<rightEdgesInt.length;i++){
+        let numberDelete2 = tabsIntWithoutEdges.indexOf(rightEdgesInt[i])
+        tabsIntWithoutEdges.splice(numberDelete2,1)
+    }
+
+    while (emptyTabCopy.length !== 0){ // in order to correctly delete buttons when user is clicking on the button with an empty cell under it
+        // i've decided to divide array with all empty cells(tabs) into multiple arrays
+
+        for (let i = 0;i<emptyTabCopy.length;i++){               // creating arrays with all left and right empty tabs
+            for (let j = 0;j<leftEdgesInt.length;j++){
+                if (emptyTabCopy[i]===leftEdgesInt[j]){
+                    arrayLeft.push(leftEdgesInt[j])            // array with only left empty tabs
+                }
+                else if (emptyTabCopy[i]===rightEdgesInt[j]){
+                    arrayRight.push(rightEdgesInt[j])
+                }
+            }
+        }
+        console.log(emptyTabCopy)
+        for (let i = 0;i<leftEdgesInt.length;i++){               // deleting right and left empty tabs (different logic would be made for them)
+            for (let j = 0;j<emptyTabCopy.length;j++){
+                if (emptyTabCopy[j]===leftEdgesInt[i]){
+                    emptyTabCopy.splice(emptyTabCopy.indexOf(emptyTabCopy[j]),1)
+                }
+                else if (emptyTabCopy[j]===rightEdgesInt[i]){
+                    emptyTabCopy.splice(emptyTabCopy.indexOf(emptyTabCopy[j]),1)
+                }
+            }
+        }
+        console.log(emptyTabCopy)
+
+        array1.push(emptyTabCopy[0])
+        for (let k =0;k<5;k++){                      // rechecking massive 5 times to make sure every number is correct (if not rechecked it might left some doubles)
+
+            array1 = array1.filter((item, index) => {          // deleting doubles
+                return array1.indexOf(item) === index
+            })
+            for (let i = 0;i<emptyTabCopy.length;i++){
+                for (let j=0;j<array1.length;j++){
+                    if (emptyTabCopy[i]===array1[j]+1 || emptyTabCopy[i]===array1[j]-board1.y || emptyTabCopy[i]===array1[j]-1 || emptyTabCopy[i]===array1[j]+board1.y){
+                        array1.push(emptyTabCopy[i])
+                    }
+
+
+                }
+            }
+        }
```

Comment:
create separate functions from above code. Each functions should be atomic do one thing not multiple things. For example `initializeMinesBoard` which will put randomly mines on the field.